packages/core/src/use-router-params.ts

```diff
@@ -0,0 +1,13 @@
+import {useRouter} from 'next/router'
+import {useRouterQuery} from './use-router-query'
+
+export function useRouterParams() {
+  const router = useRouter()
+  const query = useRouterQuery()
+
+  return Object.fromEntries(
+    Object.entries(router.query).filter(
+      ([key, value]) => typeof query[key] === 'undefined' || query[key] !== value,
+    ),
+  )
```

Comment:
This is the part that needs unit tests — you can extract it into a function outside useRouterParams, export it, and then test this.

Reply:
Yeah, for sure, that could be exported and tested. Thanks for pointing this out. I will make a unit test for that.

Reply:
@flybayer the test is added. Let's see does it pass 😄.